### PR TITLE
feat(shwap): Add validation file

### PR DESCRIPTION
**Pull request by walldiss**
Adds simple validation middleware to file to prevent unwanted access by invalid index.

### CODE REVIEW

**Comment by cristaloleg** [comment]
Commits from another PR? https://github.com/celestiaorg/celestia-node/pull/3484/commits

**Review by cristaloleg — APPROVED** [review]
SGTM

File: share/new_eds/validation.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ Accessor = validation{}
```
Comment (Wondertan) [inline share/new_eds/validation.go:14]: Should implement Closer as we discussed.

File: share/new_eds/validation.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	return &validation{Accessor: f}
+}
+
+func (f validation) Sample(ctx context.Context, rowIdx, colIdx int) (shwap.Sample, error) {
+	if err := validateIndexBounds(ctx, f, colIdx); err != nil {
+		return shwap.Sample{}, fmt.Errorf("col: %w", err)
+	}
+	if err := validateIndexBounds(ctx, f, rowIdx); err != nil {
+		return shwap.Sample{}, fmt.Errorf("row: %w", err)
+	}
```
Comment (Wondertan) [inline share/new_eds/validation.go:35]: I hope calling Size would be cheap to call it twice

Comment (walldiss) [reply share/new_eds/validation.go:43]: added caching for size just in case

File: share/new_eds/validation.go (hunk 1 above)
Comment (walldiss) [reply share/new_eds/validation.go:14]: Added WithCloser(Accessor,Closer) AccessorCloser. Very handy and will be reused later

Comment (Wondertan) [reply share/new_eds/validation.go:14]: Hmm, ok

File: share/new_eds/validation.go (hunk 2 above)
Comment (Wondertan) [reply share/new_eds/validation.go:43]: Cool, just lets make sure it concurrency safe

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
Linter break?

**Pull request merged**